src/Form/SubscriptionForm.php

```diff
@@ -83,7 +83,29 @@ public function buildForm(array $form, FormStateInterface $form_state, stdClass
     );
 
     // Build form according to received configuration:
-    // Add contact fields.
+
+    // Add form description fields as empty fieldsets.
+    // - Only use the #description attribute of fieldsets in order to show unformated,
+    //   multiline text without scrollbars or color formating.
+    // - #description text is automatically scalled to window width.
+    foreach ($profile->contact_form_descriptions as $contact_form_description_name => $contact_form_description_data) {
+      $is_active = isset($contact_form_description_data['active']) && $contact_form_description_data['active'];
+      $has_description = isset($contact_form_description_data['description']) && strlen($contact_form_description_data['description']) > 0;
+
+      if ($is_active && $has_description) {
+        $has_weight = isset($contact_form_description_data['weight'])
+                           && filter_var($contact_form_description_data['weight'], FILTER_VALIDATE_INT) !== FALSE;
+        $weight = $has_weight ? $contact_form_description_data['weight'] : 0;
+
+        $form['contact_fields'][$contact_form_description_name] = array(
+          '#type' => 'fieldset',
```

Comment:
Let's use [`markup`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Markup.php/class/Markup) instead.

Reply:
probably `[html_entity_decode(](https://www.php.net/manual/en/function.html-entity-decode.php)$contact_form_description_data['description'])` should be used.

if plain html-tags are entered into the description field on the civicrm-side (via _de.systopia.newsletter_), all those tags will be converted to their html-entity version. html-entities are not converted back when using markup.
  
if `html_entity_decode` is being called before assigning the description string to the _#markup_ field, then the plain html-tags would be recovered and then rendered correctly in the form:

`'#markup' => html_entity_decode($contact_form_description_data['description'])`
`'#markup' => Markup::create(html_entity_decode($contact_form_description_data['description']))`

testing indicates that both versions produce the same correctly rendered output when receiving html-tags like  &lt;strong&gt; or  &lt;em&gt; from the civicrm-side.

`html_entity_decode` uses **utf-8** as default for character conversion and has a default flag-argument set to `ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401`

Reply:
The API should rather send raw HTML (which requires adjusting the HTML filtering in the CiviCRM extension), so that the Drupal module does not have to decode.

Also, `Markup::create()` is the preferred method, as it implies XSS filtering of raw HTML in Drupal (e.g. stripping `<script>` tags etc.).